File: src/main/java/org/railwaystations/rsapi/core/ports/in/FindPhotoStationsUseCase.java
package org.railwaystations.rsapi.core.ports.in;

import org.railwaystations.rsapi.core.model.Station;

import java.time.Instant;
import java.util.Optional;
import java.util.Set;

public interface FindPhotoStationsUseCase {

    Set<Station> findByCountry(Set<String> countries, Boolean hasPhoto, String photographer, Boolean active);

    Set<Station> findByCountry(Set<String> countries, Boolean hasPhoto, Boolean active);

    Optional<Station> findByCountryAndId(String country, String id);

    Set<Station> findRecentImports(Instant minus);

    Set<Station> findByPhotographer(String photographer, String country);

}

File: core/src/main/kotlin/org/railwaystations/rsapi/core/ports/inbound/FindPhotoStationsUseCase.kt
package org.railwaystations.rsapi.core.ports.inbound

import org.railwaystations.rsapi.core.model.Station

interface FindPhotoStationsUseCase {
    fun findByCountry(
        countries: Set<String>,
        hasPhoto: Boolean?,
        photographer: String?,
        active: Boolean?
    ): Set<Station>

    fun findByCountry(countries: Set<String>, hasPhoto: Boolean?, active: Boolean?): Set<Station>

    fun findByCountryAndId(country: String?, stationId: String?): Station?

    fun findRecentImports(sinceHours: Long): Set<Station>

    fun findByPhotographer(photographer: String, country: String?): Set<Station>
}